pkg/config/loader_test.go

```diff
@@ -394,3 +406,111 @@ cloudquery {
 		})
 	}
 }
+
+func setupTestS3Bucket(t *testing.T) (*url.URL, *s3mem.Backend) {
+	backend := s3mem.New()
+	faker := gofakes3.New(backend)
+
+	srv := httptest.NewServer(faker.Server())
+
+	t.Cleanup(srv.Close)
+
+	assert.NoError(t, backend.CreateBucket(bucketName))
+	u, err := url.Parse(srv.URL)
+	assert.NoError(t, err)
+	return u, backend
+}
+
+func Test_LoadHCLFile(t *testing.T) {
+	srvURL, backend := setupTestS3Bucket(t)
+	cases := []struct {
+		Path        string
+		Name        string
+		Configs     string
+		Type        string
+		SetupFile   bool
+		ExpectError bool
+	}{
+		{
+			Name:      "Success-S3Object",
+			Type:      "s3",
+			Path:      fmt.Sprintf("s3://%s/config.hcl?region=us-east-1&disableSSL=true&s3ForcePathStyle=true&endpoint=%s", bucketName, srvURL.Host),
+			Configs:   testConfig,
+			SetupFile: true,
+		},
+		{
+			Name:        "Failure-S3Object",
+			Type:        "s3",
+			Path:        fmt.Sprintf("s3://%s/config2.hcl?region=us-east-1&disableSSL=true&s3ForcePathStyle=true&endpoint=%s", bucketName, srvURL.Host),
+			Configs:     testConfig,
+			SetupFile:   false,
+			ExpectError: true,
+		},
+		{
+			Name:      "Success-RelativePath",
+			Type:      "file",
+			Path:      "./asdf/asdf/asfd/teddstcdonfig.hcl",
+			Configs:   testConfig,
+			SetupFile: true,
+		},
+		{
+			Name:        "Failure-RelativePath-NotExists",
+			Type:        "file",
+			Path:        "./asdf/asdf/asfd/teddstcdonfig.hcl",
+			Configs:     testConfig,
+			SetupFile:   false,
+			ExpectError: true,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.Name, func(t *testing.T) {
+			var p *Parser
+			switch tc.Type {
+			case "s3":
+				p = NewParser()
+				os.Setenv("AWS_ANON", "true")
+				defer os.Unsetenv("AWS_ANON")
+				if tc.SetupFile {
+					assert.NoError(t, putFile(backend, tc.Path, "application/hcl", tc.Configs))
+				}
+
+			case "file":
+				appFS := afero.NewMemMapFs()
+				p = NewParser(func(p *Parser) {
+					p.fs = afero.Afero{Fs: appFS}
+				})
+				if tc.SetupFile {
+					p.fs.WriteFile(tc.Path, []byte(tc.Configs), defaultPermissions)
+				}
+			}
+			body, diags := p.LoadHCLFile(tc.Path)
```

Comment:
`LoadHCLFile` was originally `LoadConfigFile`. Since we care only about loading the file itself and not about parsing, I changed this to `LoadHCLFile` and also changed `LoadConfigFromSource` to `LoadFromSource`.

Ideally we'd only call `LoadHCLFile` and compare to a hard coded block instead of relying on `LoadFromSource`, but that requires a bigger change.

`LoadConfigFile` and `LoadConfigFromSource` are implemented in `config_parser.go` so this change keeps the test scoped to methods implemented in `loader.go`